### PR TITLE
feat(argo-workflows): Only enable PVC RBAC when persistence is enabled

### PR DESCRIPTION
**Pull request by OneCricketeer**
Seems like PVC would only be needed for the database, which is under the same condition.

Signed-off-by: Jordan Moore <1930631+OneCricketeer@users.noreply.github.com>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [ ] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [ ] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [ ] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

Changes are automatically published when merged to `main`. They are not published on branches.

### CODE REVIEW

File: charts/argo-workflows/templates/controller/workflow-controller-cluster-roles.yaml [hunk 1]
```diff
@@ -31,6 +31,7 @@ rules:
   - get
   - watch
   - list
+{{- if .Values.controller.persistence }}
 - apiGroups:
```
Comment (OneCricketeer) [inline charts/argo-workflows/templates/controller/workflow-controller-cluster-roles.yaml:35]: Or this section could be moved down into the conditional? 😅

**Comment by jmeridth** [comment]
@OneCricketeer my hesitation to move the PVC related permission fro the workflow controller under the persistence boolean is that upstream argo-workflows minimal example has them [here](https://github.com/argoproj/argo-workflows/blob/8562f95c34412a607000cb707ca7c9e61d4f9a4f/manifests/quick-start-minimal.yaml#L953) but also has quickstart examples of [mysql](https://github.com/argoproj/argo-workflows/blob/master/manifests/quick-start-mysql.yaml) and [postgresql](https://github.com/argoproj/argo-workflows/blob/master/manifests/quick-start-postgres.yaml) specifically.  I'm not 100% sure those PVCs and perms are strictly related to persistence and may still need to be exposed.  May want to ask that specific question in #argo-workflows CNCF slack.  Or maybe someone else here knows the answer.

You need an version increment and explanation of change added to the argo-workflows Chart.yaml for this PR also please (first checkbox in the description above)

**Comment by OneCricketeer** [comment]
> version increment

About that... Seems like this would be a backwards-incompatible change for anyone expecting to use PVC out of the box? So, minor bump instead of patch?

**Comment by jmeridth** [comment]
@OneCricketeer I don't think putting the PVC perms behind the persistence flag is a good idea.  The `.Values.controller.persistence` flag is if people want to use the database to offload large workflows that might exceed the 1MB limit of EtcD.  Explained [here](https://argoproj.github.io/argo-workflows/offloading-large-workflows/).  People may still want to use PVCs in their workflows like [here](https://argoproj.github.io/argo-workflows/walk-through/volumes/).

Am I missing something that you might be seeing that I'm not?

@yu-croco wdyt?

**Comment by OneCricketeer** [comment]
Ah. From docs 

> Volumes are a very useful way to move large amounts of data from one step in a workflow to another. 

Makes sense.Will close. 

Though, this does require the service account to be created, as it appears as tripped people up in the comments

**Pull request closed without merging**